python/cxx/eclipse_io.cpp

```diff
@@ -294,24 +294,27 @@ npArray get_erst_vector(Opm::EclIO::ERst * file_ptr, const std::string& key, siz
 
 
 std::tuple<std::array<double,8>, std::array<double,8>, std::array<double,8>>
-get_xyz_from_ijk(Opm::EclIO::EGrid * file_ptr,int i, int j, int k)
+get_xyz_from_ijk(Opm::EclIO::EGrid * file_ptr, int i, int j, int k)
 {
     std::array<double,8> X = {0.0};
     std::array<double,8> Y = {0.0};
     std::array<double,8> Z = {0.0};
 
-    std::array<int, 3> ijk = {i, j, k };
+    std::array<int, 3> ijk = {i, j, k};
+    
+    if (file_ptr->with_mapaxes())
+        file_ptr->getCellCornersWithMapAxes(ijk, X, Y, Z);
+    else
+        file_ptr->getCellCorners(ijk, X, Y, Z);
```

Comment:
If we are going to define separately functions, then this should be also done for `get_xyz_from_active_index` and for future functions, which might duplicate code. I think since when the EGRID file is read we set to use or not to use mapaxes, then this approach should be fine.

Reply:
> If we are going to define separately functions, then this should be also done for get_xyz_from_active_index and for future functions, which might duplicate code. 

Possibly, although the obvious alternative is that we leave the current functions as is and implement a separate function whose job it is to apply the MAPAXES transformation to a sequence of coordinates.  That way there's no duplication and the client code can get the behaviour it wants in a very explicit fashion.